### PR TITLE
runtime-rs:  support physical endpoint using device manager

### PR DESCRIPTION
**Pull request by Tim-0731-Hzt**
use the device manager to attach the physical endpoint
    
Fixes: #7103
Signed-off-by: Zhongtao Hu <zhongtaohu.tim@linux.alibaba.com>

### CODE REVIEW

File: src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs [hunk 1]
```diff
@@ -103,24 +108,19 @@ impl Endpoint for PhysicalEndpoint {
...
-            bus_mode: driver::VfioBusMode::new(mode),
-            devices: vec![HostDevice {
-                hostdev_id: format!("physical_nic_{}", self.name().await),
-                bus_slot_func: self.bdf.clone(),
-                ..Default::default()
-            }],
-            ..Default::default()
-        });
-        hypervisor.add_device(d).await.context("add device")?;
+        // create and insert block device into Kata VM
```
Comment (Apokleos) [inline src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs:119]: The comment needs to be corrected.

File: src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs [hunk 2]
```diff
@@ -94,7 +99,7 @@ impl Endpoint for PhysicalEndpoint {
         self.hard_addr.clone()
     }
 
-    async fn attach(&self, hypervisor: &dyn Hypervisor) -> Result<()> {
+    async fn attach(&self, _: &dyn Hypervisor) -> Result<()> {
```
Comment (Apokleos) [inline src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs:102]: `_hypervisor` is better than `_`.

File: src/runtime-rs/crates/resource/src/network/network_with_netns.rs [hunk 3]
```diff
@@ -149,7 +152,10 @@ impl Network for NetworkWithNetns {
...
 }
 
-async fn get_entity_from_netns(config: &NetworkWithNetNsConfig) -> Result<Vec<NetworkEntity>> {
+async fn get_entity_from_netns(
+    config: &NetworkWithNetNsConfig,
+    d: Arc<RwLock<DeviceManager>>,
+) -> Result<Vec<NetworkEntity>> {
     info!(
         sl!(),
         "get network entity for config {:?} tid {:?}",
```
Comment (Apokleos) [inline src/runtime-rs/crates/resource/src/network/network_with_netns.rs:161]: Is it "for" or "from" in the log `get network entity for config` ?

**Review by Apokleos — APPROVED** [review]
Thx @Tim-0731-Hzt 
Other than a few issues about comments, everything else is fine with me.

LGTM.

**Review by quanweiZhou — COMMENTED** [review]
Thanks, @Tim-0731-Hzt, LGTM.

File: src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs (hunk 1 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs]: thx, fixed

File: src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs (hunk 2 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/resource/src/network/endpoint/physical_endpoint.rs]: fixed

File: src/runtime-rs/crates/resource/src/network/network_with_netns.rs (hunk 3 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/resource/src/network/network_with_netns.rs]: yes, fixed

**Review by lifupan — APPROVED** [review]

**Review by quanweiZhou — APPROVED** [review]

**Pull request merged**